site/en/docs/extensions/mv3/sandboxingEval/index.md

```diff
@@ -108,32 +160,55 @@ exactly as expected, and we end up with a compiled template in `templates['hello
 ### Pass the result back {: #pass_result }
 
 We'll make this template available for use by setting up a message listener that accepts commands
-from the Event Page. We'll use the `command` passed in to determine what ought to be done (you could
+from the extension page. We'll use the `command` passed in to determine what ought to be done (you could
 imagine doing more than simply rendering; perhaps creating templates? Perhaps managing them in some
 way?), and the `context` will be passed into the template directly for rendering. The rendered HTML
 will be passed back to the Event Page so the extension can do something useful with it later on:
 
 ```html
-<script>
-  window.addEventListener('message', function(event) {
-    var command = event.data.command;
-    var name = event.data.name || 'hello';
-    switch(command) {
-      case 'render':
-        event.source.postMessage({
-          name: name,
-          html: templates[name](event.data.context)
-        }, event.origin);
-        break;
-
-      // case 'somethingElse':
-      //   ...
-    }
-  });
-</script>
+ <script>
+      const templatesElements = document.querySelectorAll(
+        "script[type='text/x-handlebars-template']"
+      );
+      let templates = {},
+        source,
+        name;
+
+      // precompile all templates in this page
+      for (let i = 0; i < templatesElements.length; i++) {
+        source = templatesElements[i].innerHTML;
+        name = templatesElements[i].id;
+        templates[name] = Handlebars.compile(source);
+      }
+
+      // Set up message event handler:
+      window.addEventListener('message', function (event) {
+        const command = event.data.command;
+        const template = templates[event.data.templateName];
+          let result = 'invalid request';
+
+        // if we don't know the templateName requested, return an error message
+        if !(template) {
```

Comment:
This is a syntax error, what I meant was:


```
if (template) {
       switch (command) {
            case 'render':
              result = template(event.data.context);
              break;
            // you could even do dynamic compilation, by accepting a command
            // to compile a new template instead of using static ones, for example:
            // case 'new':
            //   template = Handlebars.compile(event.data.templateSource);
            //   result = template(event.data.context);
            //   break;
          }
        } else {
          result = 'Unknown template: ' + event.data.templateName;
        }
```

Reply:
(with correct indentation)